Automate deletion of temporary files on windows bots

aj062 opened the pull request:
#### 9095b91f6ddaab1506f93fa0154f568cf2ebe6bf
<pre>
Automate deletion of temporary files on windows bots
<a href="https://bugs.webkit.org/show_bug.cgi?id=245786">https://bugs.webkit.org/show_bug.cgi?id=245786</a>
&lt;rdar://100635624&gt;

Reviewed by Darin Adler.

* Tools/CISupport/delete-stale-build-files:
* Tools/CISupport/ews-build/factories.py:
* Tools/CISupport/ews-build/steps.py:
(DeleteStaleBuildFiles):
(DeleteStaleBuildFiles.__init__):
* Tools/CISupport/ews-build/steps_unittest.py:
* Tools/CISupport/ews-build/factories_unittest.py:

Canonical link: <a href="https://commits.webkit.org/255619@main">https://commits.webkit.org/255619@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8fd6d489710cca59c61c53d42b127aaf206022cb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/91651 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/880 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/22254 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/101349 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/161438 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/95656 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/878 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/29486 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/83968 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/97680 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/97309 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/510 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/78306 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/27466 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/82434 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/82081 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/70504 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/35773 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/16108 "Passed tests") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/90716 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/33528 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/17204 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/37369 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/39957 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1791 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/39274 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/36321 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

roynos commented on Tools/CISupport/delete-stale-build-files:
```diff
@@ -86,6 +91,20 @@ def webkitBuildDirectory(platform, fullPlatform, configuration):
+        except OSError as e:
+            print(f'Failed to remove: {directory}, error: {e}')
+            continue
+        print('Done')
```
do you want to print Done every single time?

aj062 replied:
It doesn't look ugly as such: https://ews-build.webkit-uat.org/#/builders/13/builds/4316/steps/5/logs/stdio, but I can remove if anyone has a strong opinion on it.

aj062 committed:
Automate deletion of temporary files on windows bots
https://bugs.webkit.org/show_bug.cgi?id=245786
<rdar://100635624>

Reviewed by Darin Adler.

* Tools/CISupport/delete-stale-build-files:
* Tools/CISupport/ews-build/factories.py:
* Tools/CISupport/ews-build/steps.py:
(DeleteStaleBuildFiles):
(DeleteStaleBuildFiles.__init__):
* Tools/CISupport/ews-build/steps_unittest.py:
* Tools/CISupport/ews-build/factories_unittest.py:

Canonical link: https://commits.webkit.org/255619@main
